### PR TITLE
Stop tensorboard on sweep or experiment delete

### PR DESCRIPTION
**Pull request by lantiga**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning-hpo/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?

This PR ensures that Tensorboard works are stopped when sweeps or experiments are deleted.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: lightning_hpo/controllers/sweep.py [hunk 1]
```diff
@@ -119,23 +119,32 @@ def delete_sweep(self, config: DeleteSweepConfig):
                     sweep = sweep.collect_model()
                     del self.r[config.name]
                 self.db.delete(sweep)
+                for tensorboard in self.db.select_all(TensorboardConfig):
```
Comment (tchaton) [inline lightning_hpo/controllers/sweep.py:122]: I believe this should delete tensorboard instead and every resources associated to the Sweep.

File: lightning_hpo/controllers/sweep.py [hunk 2]
```diff
@@ -119,23 +119,32 @@ def delete_sweep(self, config: DeleteSweepConfig):
...
+                    if config.name != sweep.sweep_id:
+                        return f"The experiment `{config.name}` is part of sweep `{sweep.sweep_id}`, which includes multiple experiments. Deleting individual experiments of a sweep is currently unsupported."
+                    if config.name in self.r:
+                        sweep: Sweep = self.r[config.name]
+                        for w in sweep.works():
+                            w.stop()
+                        sweep = sweep.collect_model()
+                        del self.r[config.name]
+                    self.db.delete(sweep)
+                    for tensorboard in self.db.select_all(TensorboardConfig):
```
Comment (tchaton) [inline lightning_hpo/controllers/sweep.py:143]: Same as above.

File: lightning_hpo/controllers/sweep.py [hunk 3]
```diff
@@ -119,23 +119,32 @@ def delete_sweep(self, config: DeleteSweepConfig):
...
-                    sweep: Sweep = self.r[config.name]
-                    for w in sweep.works():
-                        w.stop()
-                    sweep = sweep.collect_model()
-                    del self.r[config.name]
-                self.db.delete(sweep)
-                return f"Deleted the experiment `{config.name}`"
+            for experiment in sweep.experiments.values():
+                if config.name == experiment.name:
+                    if config.name != sweep.sweep_id:
```
Comment (tchaton) [inline lightning_hpo/controllers/sweep.py:134]: Neat !

File: tests/controllers/test_sweep.py [hunk 4]
```diff
@@ -55,3 +58,60 @@ def test_sweep_controller(monkeypatch):
...
+    assert len(sweep_controller.db.select_all(TensorboardConfig)) == 1
+
+    delete_config = DeleteSweepConfig()
+    delete_config.name = config.sweep_id
+
+    sweep_controller.delete_sweep(delete_config)
+    assert len(sweep_controller.db.select_all(TensorboardConfig)) == 0
+
+
+def test_sweep_controller_delete_experiment(monkeypatch):
```
Comment (tchaton) [inline tests/controllers/test_sweep.py:91]: Let's add a test to delete an experiment from a sweep.

Comment (lantiga) [reply tests/controllers/test_sweep.py:93]: Done

**Pull request merged**